paddle/phi/kernels/stride/set_inplace.cc

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2024 PaddlePaddle Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#include "paddle/common/flags.h"
+#include "paddle/phi/backends/all_context.h"
+#include "paddle/phi/core/kernel_registry.h"
+#include "paddle/phi/kernels/set_inplace_kernel.h"
+
+COMMON_DECLARE_bool(use_stride_kernel);
+
+namespace phi {
+
+template <typename T, typename Context>
+void SetInplaceKernel(const Context& dev_ctx,
+                      const DenseTensor& x,
+                      const DenseTensor& source,
+                      const std::vector<int64_t>& dims,
+                      const std::vector<int64_t>& stride,
+                      int64_t offset,
+                      DenseTensor* out) {
+  if (!FLAGS_use_stride_kernel) {
+    PADDLE_THROW(common::errors::Fatal(
+        "FLAGS_use_stride_kernel is closed. Strided kernel "
+        "be called, something wrong has happened!"));
+  }
+  auto meta = out->meta();
+  meta.dims = DDim(dims.data(), static_cast<int>(dims.size()));
+  meta.strides = DDim(stride.data(), static_cast<int>(stride.size()));
+  meta.offset = offset;
+  // reset holder to nullptr
+  out->clear();
+  *out = DenseTensor{source.Holder(), meta};
+  out->ShareInplaceVersionCounterWith(x);
+}
+
+}  // namespace phi
+
+PD_REGISTER_KERNEL(set,
+                   CPU,
+                   STRIDED,
+                   phi::SetInplaceKernel,
+                   bool,
+                   uint8_t,
+                   int8_t,
+                   int16_t,
+                   int,
+                   int64_t,
+                   float,
+                   double,
+                   phi::dtype::complex<float>,
```

Comment:
is CPU unspport float16 and bfloat16？

Reply:
I don't know whether all kind of CPU support `float16` and `bfloat16` so they are not registered here

Reply:
I think you can try the same way this paddle/phi/kernels/strided_slice_kernel.cc

Reply:
OK, `float16` and `bfloat16` are added.

Reply:
`STRIDED` may change to `ALL_LAYOUT` ？